File: python/leetcode/487-max-consecutive-ones-ii.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        longest = 0
        left = 0
        zeroes = 0

        for right in range(len(nums)):

            if nums[right] == 0:
                zeroes += 1

            while zeroes > 1:
                if nums[left] == 0:
                    zeroes -= 1
                left += 1

            longest = max(longest, right-left+1)

        return longest

File: c#/leetcode/487-max-consecutive-ones-ii.cs
public class Solution
{
  public int FindMaxConsecutiveOnes(int[] nums)
  {
    int left = 0;
    int longest = 0;
    int zeroes = 0;

    for (int right = 0; right < nums.Length; right++)
    {
      if (nums[right] == 0)
      {
        zeroes++;
      }

      while (zeroes > 1)
      {
        if (nums[left] == 0)
        {
          zeroes--;
        }
        left++;
      }
      longest = Math.Max(longest, right - left + 1);
    }
    return longest;
  }
}